src/EventInvocation.js

```diff
@@ -152,8 +148,16 @@ class EventRegistrationInterface {
     }
   }
 
-  // Check and assign SDK type from incoming params
-  getSdkTypeAndModule(moduleWithEventName) {
+  /**
+   * parseEventNameAndModuleAndSDKType
+   * This method parses the module with event name to extract SDK type, module, event name without module, and formatted event name.
+   * @param {String} moduleWithEventName - The module with event name in the format 'sdkType_moduleName.onEventName'.
+   * @returns - array containing sdkType, module, eventNameWithoutModule, and formatted eventName.
+   * @example
+   * parseEventNameAndModuleAndSDKType('firebolt_foo.onExampleEvent')
+   * returns ['firebolt', 'foo', 'onExampleEvent', 'exampleEvent']
+   */
+  parseEventNameAndModuleAndSDKType(moduleWithEventName) {
```

Comment:
This is great. Can we use this in the other locations where we're taking a "moduleWithEventName" and parsing it?